qt/PrefsDialog.cc

```diff
@@ -295,6 +307,22 @@ void PrefsDialog::pathChanged(QString const& path)
     }
 }
 
+void PrefsDialog::plainTextChanged()
+{
+    PreferenceWidget const pref_widget(sender());
+
+    if (pref_widget.is<QPlainTextEdit>())
+    {
+        auto const* const plain_text_edit = pref_widget.as<QPlainTextEdit>();
+
+        if (plain_text_edit->document()->isModified())
+        {
+            prefs_.set(pref_widget.getPrefKey(), plain_text_edit->toPlainText());
+            // we avoid using setPref() because the included refreshPref() call would reset the cursor while we're editing
```

Comment:
Maybe connect to `focusChanged`? I don't have the docs in front of me so I don't know if it's doable, but if so it seems like it would be simpler

Reply:
If it ain't broken...

Reply:
Does this event get fired every time a character is changed in the text edit, or once when the user is done? E.g. what happens if someone physically types the URLs into the text edit?

Reply:
"[QTextDocument](https://doc.qt.io/qt-5/qtextdocument.html) emits a [textChanged](https://doc.qt.io/qt-5/qplaintextedit.html#textChanged)() signal if the text changes and it also provides a isModified() function which will return true if the text has been modified since it was either loaded or since the last call to setModified with false as argument." - https://doc.qt.io/qt-5/qplaintextedit.html

So yes, modifying URLs in that text box is working. There might be some internal Qt lag between typing and triggering the signal, to avoid triggering it too often.

Reply:
> So yes, modifying URLs in that text box is working.

My concern is not that zero modifications are made, but that when someone starts typing, we will flood tr_session with hundreds of tracker rebuilds.

Fixed in https://github.com/transmission/transmission/pull/2668/commits/3780f90ef554e21504e5ac2ddfe79ff4b602c68c#diff-3ebbc2c0d8da18b7c49534751edd57bb14351013218ef4e192e5287d6e0c9033R263 by comparing the text contents on focus-in + focus-out and modifying the prefs iff the two differ